pkg/watermark/fetch/processor_manager.go

```diff
@@ -249,10 +249,27 @@ func (v *ProcessorManager) startTimeLineWatcher() {
 					v.log.Errorw("Unable to decode the value", zap.String("processorEntity", p.entity.GetName()), zap.Error(err))
 					continue
 				}
-				p.offsetTimeline.Put(OffsetWatermark{
-					watermark: otValue.Watermark,
-					offset:    otValue.Offset,
-				})
+				if otValue.Idle {
+					var processors = v.GetAllProcessors()
+					for processorName, processor := range processors {
+						if processorName != value.Key() {
```

Comment:
why are we checking for this condition?

Reply:
I want to skip the processor itself.. added a comment.